server/events/working_dir.go

```diff
@@ -295,8 +254,50 @@ func (w *FileWorkspace) forceClone(log logging.SimpleLogging,
 			return fmt.Errorf("running %s: %s: %s", cmdStr, sanitizedOutput, sanitizedErrMsg)
 		}
 		log.Debug("ran: %s. Output: %s", cmdStr, strings.TrimSuffix(sanitizedOutput, "\n"))
+		return nil
+	}
+
+	if !w.CheckoutMerge {
+		return runGit("clone", "--depth=1", "--branch", p.HeadBranch, "--single-branch", headCloneURL, cloneDir)
 	}
-	return nil
+
+	if err := runGit("clone", "--depth", fmt.Sprint(w.CheckoutDepth), "--branch", p.BaseBranch, "--single-branch", baseCloneURL, cloneDir); err != nil {
```

Comment:
How about removing `--depth` option when `CheckoutDepth` is 0 and default set 0 to keep the original behavior? `CheckoutDepth` can be optional for large-mono repo users.

`runGit("fetch", "--depth", fmt.Sprint(w.CheckoutDepth), fetchRemote, fetchRef)`, too.

Reply:
> How about removing --depth option when CheckoutDepth is 0

See this https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt

> --depth <depth>
>
>    Create a shallow clone with a history truncated to the specified number of commits. Implies --single-branch unless --no-single-branch is given to fetch the histories near the tips of all branches. If you want to clone submodules shallowly, also pass --shallow-submodules

The benefit of always specifying the depth is that we only come a single branch instead of all the branches which speeds up clone time for both branch and merge check out strategies

Reply:
> and default set 0 to keep the original behavior? CheckoutDepth can be optional for large-mono repo users.

This is a good point. I can set the default to 0.

Reply:
I think it would be good to specify default depth as 1, and that should be shadow clone rather than 0 (which is full clone)? (That is also a default checkout behavior for checkout action?)